test/e2e/upgrade_test.go

```diff
@@ -37,6 +37,15 @@ func runSimpleUpgradeFlow(test *framework.ClusterE2ETest, updateVersion v1alpha1
 	test.DeleteCluster()
 }
 
+func runUpgradeFlowWithAPI(test *framework.ClusterE2ETest, updateVersion v1alpha1.KubernetesVersion, clusterOpts ...framework.ClusterE2ETestOpt) {
+	test.GenerateClusterConfig()
+	test.CreateCluster()
+	test.UpgradeClusterWithKubectl(clusterOpts)
+	test.ValidateCluster(updateVersion)
```

Comment:
Do we still need this after using `ValidateClusterState`?

Reply:
`ValidateCluster` also validates the node version. So I can add that to both of the node validation methods & remove this